docs/root/configuration/http/http_filters/gzip_filter.rst

```diff
@@ -45,7 +45,8 @@ By *default* compression will be *skipped* when:
   that the "gzip" will have a higher weight then "\*". For example, if *accept-encoding*
   is "gzip;q=0,\*;q=1", the filter will not compress. But if the header is set to
   "\*;q=0,gzip;q=1", the filter will compress.
-- A request whose *accept-encoding* header includes "identity".
+- A request whose *accept-encoding* header includes any encoding type with a higher
+  weight than "gzip"'s.
```

Comment:
Yes, I try the highest weighted one.

I' m going to clarify a bit this statement by updating it to
> A request whose *accept-encoding* header includes any encoding type with a higher weight than "gzip"'s given the corresponding compression filter is present in the chain.

But now I see an issue: let's say the "gzip" filter is configured to compress text only and "brotli" to compress JPEGs only, then a request comes with "Accept-Encoding: gzip;q=1, br;q=.5" and the response contains "Content-Type: image/jpg". In this case no compression would be applied at all with the current implementation.

I guess the decision on which filter to use should be delayed until "Content-Type" is known. And it should be done in the scope of this PR, because it will become important when "brotli" is added to Envoy.

Reply:
@jmarantz PTAL. Now the decision which filter to use is delayed until content-type is known.